src/app/modules/item/pages/item-display/item-display.component.ts

```diff
@@ -140,26 +148,32 @@ export class ItemDisplayComponent implements OnInit, AfterViewInit, OnDestroy {
 }
 
 export class ItemDisplayPlatform extends Platform {
-  taskParams: any;
+  taskParams: TaskParams;
   constructor(task: Task, taskParams: TaskParams) {
     super(task);
     this.taskParams = taskParams;
   }
 
-  validate(mode : string, success : CompleteFunction, _error : ErrorFunction) : void {
+  validate(mode : string) : Observable<void> {
```

Comment:
if it is the only valid values:
```suggestion
  validate(mode : 'validate'|'cancel') : Observable<void> {
```

Reply:
No there can be other values too, there are a lot of possible ones and the ones not recognized by AlgoreaFrontend should just be ignored.